tests/hmac.rs

```diff
@@ -41,7 +42,9 @@ fn encode_with_custom_header() {
         company: "ACME".to_string(),
         exp: OffsetDateTime::now_utc().unix_timestamp() + 10000,
     };
-    let header = Header { kid: Some("kid".to_string()), ..Default::default() };
+    let mut extra = HashMap::with_capacity(1);
```

Comment:
I think we should have 3 tests:
- None
- 1 extra header
- 2 or more extra headers